Add footer feature

haywirebanana opened the pull request:
Created the Footer, but removed the "sign up to present" button. 
(Idea: User inputs email to subscribe?)

![image](https://github.com/user-attachments/assets/9fa31a9d-dbd7-4175-a757-e24752f1ead7)

NobleTyler committed:
Update README.md

NobleTyler committed:
Add pnpm to readme

NobleTyler committed:
Update README.md

lindan4 committed:
Created file.

lindan4 committed:
Rebased and updated file.

lindan4 committed:
Added styling for anchor elements, but still need to handle styling for last element.

lindan4 committed:
Added border and border radius.

lindan4 committed:
Added code to cloud icon as svg and to nav bar.

lindan4 committed:
Added column gutter and animation.

haywirebanana committed:
needs css fixes

haywirebanana committed:
needs css fixes

lindan4 committed:
Upgraded Astro.

victoriacheng15 committed:
add header component

victoriacheng15 committed:
add code to cloud logo and background for the header

haywirebanana committed:
Adding Footer

haywirebanana committed:
Adding Footer

haywirebanana committed:
Fixed Body padding&margins

haywirebanana committed:
Fixed Body padding&margins

lindan4 commented on src/pages/index.astro:
```diff
@@ -61,6 +62,9 @@ import Card from '../components/Card.astro';
 			/>
 		</ul>
 	</main>
+	<footer>
+		<Footer />
+	</footer>
```
I have one qualm with the name of the component. How does FooterContent possible sound?

haywirebanana replied:
I agree! This will help with fixing problems as they're easily differentiated with each other.

lindan4 committed:
Updated README.md

What did we do?

Update documentation by replacing npm commands with pnpm commands

Why was it done?

Since we are using pnpm, different files will be generated.

lindan4 committed:
Upgraded Astro to 5.1.2

What was done?

Ran pnpm dlx @astrojs/upgrade tp upgrade Astro to the latest version.

why was this done?

To keep Astro up to date. This is major version upgrade, and considering how early we are in the development of the project, it is better to do it now.

haywirebanana committed:
removed package-lock.json -- using pnpm-lock.yaml

haywirebanana committed:
removed package-lock.json -- using pnpm-lock.yaml

haywirebanana committed:
renaming "Footer" to "FooterContent"

haywirebanana committed:
renaming "Footer" to "FooterContent"

haywirebanana committed:
renaming components/Footer.astro to components/FooterContent.astro:

haywirebanana committed:
renaming components/Footer.astro to components/FooterContent.astro:

lindan4 committed:
Merge pull request #27 from codetocloudinc/feature/nav-bar

Code to Cloud nav bar

lindan4 committed:
Merge pull request #30 from codetocloudinc/add-header

add hero component

lindan4 commented on src/layouts/Layout.astro:
```diff
@@ -13,6 +14,7 @@ const { title } = Astro.props;
 		<meta name="description" content="Astro description" />
 		<meta name="viewport" content="width=device-width" />
 		<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+		<link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css" rel="stylesheet">
```
Why do we need this?

haywirebanana replied:
I had this before the change to pnpm and didn't remove it after some testing! It will be removed in the next commit

haywirebanana replied:
This link was used to fetch icons from an external dependency (Font Awesome). Would it be cleaner to use PNGs for icons instead, or is there a way to install Font Awesome (or similar icon libraries) as a dependency via pnpm?

lindan4 commented on src/components/FooterContent.astro:
```diff
@@ -0,0 +1,150 @@
+<div class="footer-all">
+  <div class="top-footer">
+    <h1>Collaborate with us</h1>
+    <p>Interested in further connecting with us?<br>Engage through our socials, or leave us a message below.</p>
```
What do you say to putting the text here in a div with 2 p elements instead of using a br tag?

haywirebanana replied:
Done in the next commit!

lindan4 commented:
@haywirebanana you will have to rebase this PR as there are merge conflicts.

lindan4 reviewed (changes requested):
@haywirebanana you will have to rebase this PR as there are merge conflicts.

haywirebanana committed:
adjusted footer-header and footer-description

haywirebanana committed:
adjusted footer-header and footer-description

haywirebanana committed:
Merge branch 'feature/footer' of https://github.com/codetocloudinc/code-to-cloud-home into feature/footer
